File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/SpectatorJoined.java
package tech.zolhungaj.amqapi.servercommands.gameroom;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.Command;
import tech.zolhungaj.amqapi.servercommands.CommandTypeOld;
import java.util.Optional;

public record SpectatorJoined(
        @Json(name = "name") String playerName,
        Optional<Integer> gamePlayerId //Usually null, might be set in ranked??
) implements Command {
    @Override
    public String commandName() {
        return CommandTypeOld.NEW_SPECTATOR.commandName;
    }
}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/SpectatorJoined.kt
package tech.zolhungaj.amqapi.servercommands.gameroom

import com.squareup.moshi.Json
import tech.zolhungaj.amqapi.servercommands.CommandType

@JvmRecord
@CommandType("New Spectator")
data class SpectatorJoined(
    @Json(name = "name")
    val playerName: String,
    val gamePlayerId: Int? //Usually null, might be set in ranked??
)
